models/gif_psc_exp.h

```diff
@@ -0,0 +1,425 @@
+/*
+ *  gif_psc_exp.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef gif_psc_exp_H
+#define gif_psc_exp_H
+
+// Includes from nestkernel:
+#include "event.h"
+#include "archiving_node.h"
+#include "ring_buffer.h"
+#include "connection.h"
+#include "universal_data_logger.h"
+
+#include "nest.h"
+#include "poisson_randomdev.h"
+#include "gamma_randomdev.h"
+
+/* BeginDocumentation
+  Name: gif_psc_exp - Current based generalized integrate-and-fire neuron model
+  according to Mensi et al. (2012)
+  and Pozzorini et al. (2015)
+
+  Description:
+
+  gif_psc_exp is the generalized integrate-and-fire neuron according to Mensi et
+  al. (2012)
+  and Pozzorini et al. (2015), with exponential shaped postsynaptic currents.
+
+  This model features both an adaptation current and a dynamic threshold for
+  spike-frequency
+  adaptation. The membrane potential (V) is described by the differential
+  equation:
+
+  C*dV(t)/dt = -g_L*(V(t)-E_L) - eta_1(t) - eta_2(t) - ... - eta_n(t) + I(t)
+
+  where each eta_i is a spike triggered current (stc), and the neuron model can
+  have arbitrary number of them.
+  Dynamic of each eta_i is described by:
+
+  Tau_eta_i*d{eta_i}/dt = -eta_i
+
+  and in case of spike emission, its value increased by a constant (which can be
+  positive or negative):
+
+  eta_i = eta_i + q_eta_i  (in case of spike emission).
+
+  Neuron produces spikes STOCHASTICALLY according to a point process with the
+  firing intensity:
+
+  lambda(t) = lambda0 * exp[(V(t)-V_T(t)/delta_u)]
+
+  where V_T(t) is a time-dependent firing threshold:
+
+  V_T(t) = V_T_star + gamma_1(t) + gamma_2(t) + ... + gamma_m(t)
+
+  where gamma_i is a kernel of spike-frequency adaptation (sfa), and the neuron
+  model can have arbitrary number of them.
+  Dynamic of each gamma_i is described by:
+
+  Tau_gamma_i*d{gamma_i}/dt = -gamma_i
+
+  and in case of spike emission, its value increased by a constant (which can be
+  positive or negative):
+
+  gamma_i = gamma_i + q_gamma_i  (in case of spike emission).
+
+  In the source code and parameter names we use stc and sfa, respectively
+  instead of eta and gamma.
```

Comment:
This information does not really belong into the user space documentation, since it is only relevant for developers.